localization/autoware_localization_evaluation_scripts/scripts/plot_diagnostics.py

```diff
@@ -0,0 +1,302 @@
+#!/usr/bin/env python3
+"""A script to parse diagnostics messages from a rosbag file."""
+
+import argparse
+from pathlib import Path
+
+from builtin_interfaces.msg import Time
+import matplotlib.pyplot as plt
+import pandas as pd
+from rclpy.serialization import deserialize_message
+import rosbag2_py
+from rosidl_runtime_py.utilities import get_message
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser()
+    parser.add_argument("rosbag_path", type=Path)
+    parser.add_argument("--save_dir", type=Path, default=None)
+    return parser.parse_args()
+
+
+def parse_stamp(stamp: Time) -> int:
+    return stamp.sec * int(1e9) + stamp.nanosec
+
+
+def diag_name_to_filename(diag_name: str) -> str:
+    return diag_name.replace(":", "_").replace(" ", "_")
+
+
+def parse_diagnostics_msgs(rosbag_path: str, target_list: list) -> dict:
+    serialization_format = "cdr"
+    storage_options = rosbag2_py.StorageOptions(
+        uri=str(rosbag_path),
+        storage_id="sqlite3",
```

Comment:
The bag format is mcap in the next LTS jazzy, so I would like to have a selectable method here instead of fixed `sqlite3`.

Reply:
def parse_diagnostics_msgs(rosbag_path: str, target_list: list, storage_type: str) -> dict:
    serialization_format = "cdr"
    storage_options = rosbag2_py.StorageOptions(
        uri=str(rosbag_path),
        storage_id=storage_type,

Reply:
I used mcap bag, and get the error message below.

```
❯ ros2 run autoware_localization_evaluation_scripts plot_diagnostics.py /home/hayatomizushima/dlr2_data/localization/out/latest/result_bag
[ERROR] [1740619363.822279485] [rosbag2_storage]: Could not open '/home/hayatomizushima/dlr2_data/localization/out/latest/result_bag/result_bag_0.mcap' with 'sqlite3'. Error: Failed to setup storage. Error: Error when processing SQL statement. SQLite error (26): file is not a database
[ERROR] [1740619363.822291104] [rosbag2_storage]: Could not load/open plugin with storage id 'sqlite3'
```